### PR TITLE
feature: correct counts for the search

### PR DESCRIPTION
**Pull request by densumesh**

### CODE REVIEW

File: server/src/operators/search_operator.rs [hunk 1]
```diff
@@ -237,14 +239,38 @@ pub async fn retrieve_qdrant_points_query(
...
+    let (point_ids, count) = futures::join!(point_ids_future, count_future);
     transaction.finish();
 
+    let pages = (count.map_err(|e| {
+        log::error!("Failed to get point count from Qdrant {:?}", e);
+        DefaultError {
+            message: "Failed to get point count from Qdrant",
+        }
+    })? as f64
+        / 10.0)
```
Comment (skeptrunedev) [inline server/src/operators/search_operator.rs:262]: I think this should be `limit`

File: server/src/operators/search_operator.rs [hunk 2]
```diff
@@ -237,14 +239,38 @@ pub async fn retrieve_qdrant_points_query(
...
+        limit,
+        score_threshold,
+        vector,
+        config.clone(),
+    );
+
+    let count_future = get_point_count_qdrant_query(filter, config);
+
+    let (point_ids, count) = futures::join!(point_ids_future, count_future);
     transaction.finish();
```
Comment (skeptrunedev) [inline server/src/operators/search_operator.rs:254]: Should remove this

File: server/src/operators/search_operator.rs [hunk 3]
```diff
@@ -237,14 +239,38 @@ pub async fn retrieve_qdrant_points_query(
...
+
     Ok(SearchChunkQueryResult {
-        search_results: point_ids.clone(),
+        search_results: point_ids.map_err(|e| {
+            log::error!("Failed to get point count from Qdrant {:?}", e);
+            DefaultError {
+                message: "Failed to get point count from Qdrant",
+            }
+        })?,
         //FIXME: dont have total results now
```
Comment (skeptrunedev) [inline server/src/operators/search_operator.rs:272]: Remove the `FIXME` comment

File: server/src/operators/search_operator.rs [hunk 4]
```diff
@@ -278,20 +304,37 @@ pub async fn retrieve_group_qdrant_points_query(
...
+
+    let (point_ids, count) = futures::join!(point_id_future, count_future);
+
+    let pages = (count.map_err(|e| {
+        log::error!("Failed to get point count from Qdrant {:?}", e);
+        DefaultError {
+            message: "Failed to get point count from Qdrant",
+        }
+    })? as f64
+        / 10.0)
```
Comment (skeptrunedev) [inline server/src/operators/search_operator.rs:327]: Divide by `limit`

File: server/src/operators/search_operator.rs [hunk 5]
```diff
@@ -404,19 +447,36 @@ pub async fn search_within_chunk_group_query(
...
+
+    let (point_ids, count) = futures::join!(point_ids_future, count_future);
+
+    let pages = (count.map_err(|e| {
+        log::error!("Failed to get point count from Qdrant {:?}", e);
+        DefaultError {
+            message: "Failed to get point count from Qdrant",
+        }
+    })? as f64
+        / 10.0)
```
Comment (skeptrunedev) [inline server/src/operators/search_operator.rs:469]: Divide by `limit`

File: server/src/operators/qdrant_operator.rs [hunk 6]
```diff
@@ -946,3 +946,31 @@ pub async fn recommend_qdrant_groups_query(
         .collect();
     Ok(recommended_point_ids)
 }
+
```
Comment (fastfists) [inline server/src/operators/qdrant_operator.rs:949]:
```suggestion
#[tracing::instrument]
```

**Review by skeptrunedev — CHANGES_REQUESTED** [review]

**Review by fastfists — CHANGES_REQUESTED** [review]
Just add in the tracing for the function

File: server/src/operators/search_operator.rs (hunk 2 above)
Comment (densumesh) [reply server/src/operators/search_operator.rs:254]: Remove which thing?

**Review by skeptrunedev — APPROVED** [review]

**Pull request merged**